### PR TITLE
add rust and js lints

### PR DESCRIPTION
**Pull request by Aursen**

### CODE REVIEW

**Comment by mschneider** [comment]
can you push a complete prettier config file and lint github action? this way formatting changes wont be an issue going forward

**Comment by Aursen** [comment]
Do you want also a CI lint check ?

**Comment by mschneider** [comment]
yeah ideally, used this one before https://github.com/blockworks-foundation/mango-ui-v3/blob/main/.github/workflows/node.js.yml

**Comment by Aursen** [comment]
Ok I added but I cannot test the workflow

**Comment by mschneider** [comment]
kicked it off

File: dex/crank/src/lib.rs [hunk 1]
```diff
@@ -1592,5 +1591,6 @@ async fn read_queue_length_loop(
         format!("{{ \"length\": {}  }}", len)
     });
 
-    Ok(warp::serve(get_data).run(([127, 0, 0, 1], port)).await)
+    warp::serve(get_data).run(([127, 0, 0, 1], port)).await;
```
[the remark this answers is not in the document]
Comment (Aursen) [reply dex/crank/src/lib.rs:1594]: Yeah I forgot sorry

[the remark this answers is not in the document]
Comment (Aursen) [reply dex/crank/src/lib.rs:1594]: Apparently this is not a Result

[the remark this answers is not in the document]
Comment (mschneider) [reply dex/crank/src/lib.rs:1595]: oh lol, ok

**Comment by Aursen** [comment]
Do you want me to put rust fmt and clippy in the tests workflow or I separate in 3: tests, Rust lint, Js lint

**Comment by Aursen** [comment]
I hope it will be good now

**Comment by mschneider** [comment]
sorry, but i think you have a syntax error in one of the yaml files

**Comment by Aursen** [comment]
Sorry I will fix

**Comment by silas-x** [comment]
Was going to have a look at these jobs today, but good to see this is in hand! @Aursen, given the use case for prettier is about 5% of the codebase in the test files, suggest splitting into a separate workflow that is only triggered when changing the tests

**Comment by Aursen** [comment]
I need some help, I'm really not an expert with GitHub actions

**Comment by silas-x** [comment]
No worries! Lots of good fmt/clippy fixes in this PR. Suggest removing the prettier step and merge the rest? Can pickup tmw.

**Comment by Aursen** [comment]
Normally everything is good

**Comment by mschneider** [comment]
hooray!

**Review by mschneider — APPROVED** [review]

**Pull request merged**